core/app/models/spree/line_item.rb

```diff
@@ -118,14 +116,23 @@ def options=(options = {})
 
       assign_attributes options
 
-      # There's no need to call a pricer if we'll set the price directly.
+      # When price is part of the options we are not going to fetch
+      # it from the variant. Please note that this always allows to set
+      # a price for this line item, even if there is no existing price
+      # for the associated line item in the order currency.
       unless options.key?(:price) || options.key?('price')
-        self.money_price = variant.price_for(pricing_options)
+        self.money_price = price_from_options
       end
     end
 
-    def pricing_options
```

Comment:
I reverted this change for now. I'm probably missing something but at the moment I cannot see any scenario when something else then the line item itself could need to be aware of these specific line item pricing options. That's why I'd move it to the private section, but I'm ok leaving it in the public one as well.